applications/GeoMechanicsApplication/tests/cpp_tests/test_dgeosettlement.cpp

```diff
@@ -0,0 +1,118 @@
+#include "custom_workflows/dgeosettlement.h"
+#include "input_utility_stub.h"
+
+using namespace Kratos;
+
+namespace Kratos::Testing {
+
+const std::string parameter_json_settings = "{"
+                                            "\"solver_settings\":"
+                                            "{"
+                                            "\"model_part_name\":\"test\","
+                                            "\"model_import_settings\":"
+                                            "{"
+                                            "\"input_type\": \"mdpa\","
+                                            "\"input_filename\": \"mesh_stage1\""
+                                            "},"
+                                            "\"material_import_settings\": "
+                                            "{"
+                                            "\"materials_filename\": \"MaterialParameters1.json\""
+                                            "}"
+                                            "}"
+                                            "}"; // these have material_import settings
+
+
+void RunStage(KratosGeoSettlement& rSettlement) {
+    rSettlement.RunStage("",
+                         "",
+                         [](const char *) {}, // kept empty as a stub method
+                        [](const double) {}, // kept empty as a stub method
+                        [](const char *) {}, // kept empty as a stub method
+                        []() { return true; });
+}
+
+void ExpectNumberOfReadCallsIsEqualToOne(const KratosGeoSettlement &rSettlement) {
+    const auto input_utility_from_settlement = dynamic_cast<const InputUtilityStub*>(rSettlement.GetInterfaceInputUtility());
+    KRATOS_EXPECT_NE(input_utility_from_settlement, nullptr);
+    KRATOS_EXPECT_EQ(input_utility_from_settlement->NumberOfReadCalls(), 1);
+}
+
+void ExpectNumberOfMaterialCallsEqualTo(const int expectedNumberOfMaterialCalls, const KratosGeoSettlement& rSettlement) {
+    const auto input_utility_from_settlement = dynamic_cast<const InputUtilityStub*>(rSettlement.GetInterfaceInputUtility());
+    KRATOS_EXPECT_NE(input_utility_from_settlement, nullptr);
+    KRATOS_EXPECT_EQ(input_utility_from_settlement->NumberOfMaterialCalls(), expectedNumberOfMaterialCalls);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CreatingKratosGeoSettlementDoesNotThrow, KratosGeoMechanicsFastSuite) {
+    auto input_utility = std::make_unique<InputUtilityStub>(parameter_json_settings);
+
+    bool has_thrown = false;
+    try
+    {
+        KratosGeoSettlement settlement(std::move(input_utility));
+    }
+    catch (...)
+    {
+        has_thrown = true;
+    }
+
+    KRATOS_EXPECT_FALSE(has_thrown); // No other way to check that the constructor does not throw
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RunStageMakesRelevantCallsOnce, KratosGeoMechanicsFastSuite) {
+    auto input_utility = std::make_unique<InputUtilityStub>(parameter_json_settings); // invalid after the constructor
```

Comment:
The comment provides a good reason to use a temporary here :-) (i.e. avoid the `std::move`).